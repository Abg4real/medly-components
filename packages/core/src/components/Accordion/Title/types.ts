import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp{
    color?: string;
    isActive?: boolean;
    onToggle?: () => void;
    bgColor?: string;

}


