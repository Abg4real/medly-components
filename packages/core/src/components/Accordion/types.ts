import { Props as TitleProps } from './Title';

export interface Props {
    Title?: React.SFC<TitleProps>;
    Content?: React.SFC;
}