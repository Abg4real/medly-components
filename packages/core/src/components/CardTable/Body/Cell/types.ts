import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    align: 'left' | 'center' | 'right';
    wrapText: false | true;
}
