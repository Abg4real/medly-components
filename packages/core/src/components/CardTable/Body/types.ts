import { ColumnConfig, Data } from '../types';

export interface Props {
    data: Data[];
    columns: ColumnConfig[];
    onRowClick?: (rowData: object) => void;
    rowClickDisableKey?: string;
    uniqueKeyName?: string;
    withWhiteBackground?: boolean;
}
