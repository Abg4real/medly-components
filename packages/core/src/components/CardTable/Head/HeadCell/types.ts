import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    withWhiteBackground: boolean;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    isActive: boolean;
    withWhiteBackground: boolean;
    flex: number;
    align: 'left' | 'center' | 'right';
}
