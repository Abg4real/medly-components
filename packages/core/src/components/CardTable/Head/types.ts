import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    withWhiteBackground: boolean;
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
