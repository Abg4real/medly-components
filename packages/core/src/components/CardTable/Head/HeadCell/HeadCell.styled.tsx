import { SvgIcon } from '@medly-components/icons';
import { styled } from '@medly-components/utils';
import Text from '../../../Text';
import { StyledProps } from './types';

export const HeadCell = styled('th').attrs(({ withWhiteBackground }: StyledProps) => ({
    bgTheme: withWhiteBackground ? 'lightBackground' : 'darkBackground'
}))<StyledProps & { bgTheme: 'lightBackground' | 'darkBackground' }>`
    flex: ${({ flex }) => flex};
    padding: 0;
    padding: 2.4rem 2.4rem;
    display: flex;
    flex-direction: row;
    align-items: center;
    position: relative;
    justify-content: ${({ align }) => (align === 'left' ? 'flex-start' : align === 'right' ? 'flex-end' : 'center')};

    &::after {
        content: '';
        width: 100%;
        display: block;
        position: absolute;
        bottom: 0;
        left: 0;
        background-color: ${({ theme, bgTheme, isActive }) => (isActive ? theme.cardTable[bgTheme].headerActiveColor : 'transparent')};
        height: 0.2rem;
        border-radius: 0.2rem 0.2rem 0 0;
    }

    &:hover {
        ${Text.Style} {
            color: ${({ theme, bgTheme }) => theme.cardTable[bgTheme].headerHoverColor};
        }
        &::after {
            background-color: ${({ theme, bgTheme }) => theme.cardTable[bgTheme].headerHoverColor};
        }

        ${SvgIcon} {
            * {
                fill: ${({ theme, bgTheme }) => theme.cardTable[bgTheme].headerHoverColor};
            }
        }
    }

    ${Text.Style} {
        display: inline;
        cursor: default;
        color: ${({ theme, bgTheme, isActive }) =>
            isActive ? theme.cardTable[bgTheme].headerActiveColor : theme.cardTable[bgTheme].headerColor};
    }

    ${SvgIcon} {
        cursor: pointer;
        padding-left: 1rem;
        * {
            fill: ${({ theme, bgTheme, isActive }) =>
                isActive ? theme.cardTable[bgTheme].headerActiveColor : theme.cardTable[bgTheme].headerColor};
        }
    }
`;
HeadCell.defaultProps = { flex: 1, align: 'left' };
