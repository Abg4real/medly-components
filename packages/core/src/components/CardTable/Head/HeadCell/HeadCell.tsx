import { ChevronDownIcon, ChevronUpIcon, DropdownIcon } from '@medly-components/icons';
import { WithStyle } from '@medly-components/utils';
import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
import Text from '../../../Text';
import * as Styled from './HeadCell.styled';
import { Props } from './types';

export const HeadCell: FC<Props> & WithStyle = React.memo(({ defaultSortOrder, sortField, onSortChange, withWhiteBackground, column }) => {
    const { field, sortable, title, align } = column,
        [sortState, setSortState] = useState<'none' | 'asc' | 'desc'>(defaultSortOrder);

    useEffect(() => {
        sortField !== field && setSortState('none');
    }, [sortField]);

    const handleSort = useCallback(() => {
        const order = sortState === 'asc' ? 'desc' : 'asc';
        setSortState(order);
        onSortChange(field, order);
    }, [sortState, onSortChange, field]);

    const sortIcon = useMemo(
        () =>
            sortField !== field ? (
                <DropdownIcon size="S" />
            ) : sortState === 'desc' ? (
                <ChevronDownIcon size="M" />
            ) : (
                <ChevronUpIcon size="M" />
            ),
        [handleSort, sortField, field, sortState]
    );

    return (
        <Styled.HeadCell isActive={sortField === field} {...{ align, sortable, withWhiteBackground }}>
            <Styled.HeadCellContent onClick={sortable && handleSort}>
                <Text uppercase textVariant="h5">
                    {title}
                </Text>
                {sortable && sortIcon}
            </Styled.HeadCellContent>
        </Styled.HeadCell>
    );
});
HeadCell.defaultProps = {
    defaultSortOrder: 'none'
};
HeadCell.displayName = 'HeadCell';
HeadCell.Style = Styled.HeadCell;
