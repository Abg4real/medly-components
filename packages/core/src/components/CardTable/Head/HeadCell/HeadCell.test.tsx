import { fireEvent, render } from '@test-utils';
import React from 'react';
import { ColumnConfig } from '../../types';
import { HeadCell } from './HeadCell';

const dummyColumn: ColumnConfig = { title: 'Demo', field: 'demo', sortable: true },
    renderer = ({ withWhiteBackground = false, sortField = 'random', onSortChange = jest.fn(), column = dummyColumn }) =>
        render(
            <table>
                <thead>
                    <tr>
                        <HeadCell
                            sortField={sortField}
                            onSortChange={onSortChange}
                            withWhiteBackground={withWhiteBackground}
                            column={column}
                        />
                    </tr>
                </thead>
            </table>
        );

describe('Card Table HeadCell', () => {
    it('should render properly with dark background and dropdown icon by default', () => {
        const { container } = renderer({ withWhiteBackground: false, column: { title: 'Demo', field: 'demo' } });
        expect(container).toMatchSnapshot();
    });

    it('should render properly with light background and dropdown icon by default', () => {
        const { container } = renderer({ withWhiteBackground: true });
        expect(container).toMatchSnapshot();
    });

    it('should render properly with hover state', () => {
        const { container, getByText } = renderer({ withWhiteBackground: true });
        fireEvent.mouseEnter(getByText('Demo'));
        expect(container).toMatchSnapshot();
    });

    it('should call onSortChange with asc order when clicked on sorting icon', () => {
        const mockOnSortChange = jest.fn(),
            { container } = renderer({ onSortChange: mockOnSortChange, sortField: 'demo' });

        fireEvent.click(container.querySelectorAll('svg')[0]);
        expect(container).toMatchSnapshot();
        expect(mockOnSortChange).toHaveBeenCalledWith('demo', 'asc');
    });

    it('should call onSortChange with desc order when clicked on sorting icon', () => {
        const mockOnSortChange = jest.fn(),
            { container } = renderer({ onSortChange: mockOnSortChange, sortField: 'demo' });

        fireEvent.click(container.querySelectorAll('svg')[0]);
        fireEvent.click(container.querySelectorAll('svg')[0]);
        expect(container).toMatchSnapshot();
        expect(mockOnSortChange).toHaveBeenCalledWith('demo', 'desc');
    });

    test.each(['left', 'center', 'right'])('should be able to %p align ', (align: 'left' | 'center' | 'right') => {
        const column: ColumnConfig = {
                align,
                title: 'Column 1',
                field: 'column1'
            },
            { container } = renderer({ column });
        expect(container).toMatchSnapshot();
    });
});
