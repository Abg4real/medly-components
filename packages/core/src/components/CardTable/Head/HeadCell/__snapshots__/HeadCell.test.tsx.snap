// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Table HeadCell should be able to "center" align  1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Column 1
          </h5>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should be able to "left" align  1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Column 1
          </h5>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should be able to "right" align  1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Column 1
          </h5>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should call onSortChange with asc order when clicked on sorting icon 1`] = `
.c3 {
  font-size: 2.4rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c3 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Demo
          </h5>
          <svg
            class="c3"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.706 12.586l2.59-2.59a.996.996 0 0 1 1.41 0l2.59 2.59c.63.63.18 1.71-.71 1.71h-5.18c-.89 0-1.33-1.08-.7-1.71z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should call onSortChange with desc order when clicked on sorting icon 1`] = `
.c3 {
  font-size: 2.4rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c3 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Demo
          </h5>
          <svg
            class="c3"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.706 11.414l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71h-5.18c-.89 0-1.33 1.08-.7 1.71z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should render properly with dark background and dropdown icon by default 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Demo
          </h5>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should render properly with hover state 1`] = `
.c3 {
  font-size: 1.6rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c3 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Demo
          </h5>
          <svg
            class="c3"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Card Table HeadCell should render properly with light background and dropdown icon by default 1`] = `
.c3 {
  font-size: 1.6rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c3 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  padding: 2.4rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0::after {
  content: '';
  width: 100%;
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  background-color: transparent;
  height: 0.2rem;
  border-radius: 0.2rem 0.2rem 0 0;
}

.c0 .c1 {
  display: inline;
  cursor: default;
}

.c0 .sc-AykKE {
  cursor: pointer;
  padding-left: 1rem;
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          <h5
            class="c1 c2"
          >
            Demo
          </h5>
          <svg
            class="c3"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
